roaring/roaring.go

```diff
@@ -156,6 +161,21 @@ func NewBitmap(a ...uint64) *Bitmap {
 	return b
 }
 
+// NewSliceBitmap makes a new bitmap, explicitly selecting the slice containers
+// type, which performs better in cases where we expect a contiguous block of
+// containers added in ascending order, such as when extracting a range from
+// another bitmap.
+func NewSliceBitmap(a ...uint64) *Bitmap {
+	b := &Bitmap{
+		Containers: newSliceContainers(),
+	}
+	// TODO: We have no way to report this. We aren't in a server context
+	// so we haven't got a logger, nothing is checking for nil returns
+	// from this...
```

Comment:
since the bitmap's OpWriter is nil, `AddN` won't ever return a non-nil error (with the code as it is now), so this is semi-ok... would be nice to prevent future possible mishaps though. Maybe it should panic on error since that *should* be an impossible scenario?

Reply:
Huh. Hadn't thought of that. I'm inclined to leave it for now, I think.